### PR TITLE
Adding PCA rest calls and local projections

### PR DESCRIPTION
**Pull request by mmerce**

### CODE REVIEW

File: docs/index.rst [hunk 1]
```diff
@@ -3156,31 +3156,151 @@ The JSON structure for an Fusion is:
     }
 
 
-Whizzml Resources
+PCAs
+----
+
+A PCA (Principal Component Analysis) resource fits a number of orthogonal
+projects (components) to maximally capture the variance in a dataset. This
```
Comment (jaor) [inline docs/index.rst]: projections

File: docs/index.rst [hunk 2]
```diff
@@ -4772,8 +4961,10 @@ You can instantiate a local version of a remote model.
 This will retrieve the remote model information, using an implicitly built
 ``BigML()`` connection object (see the ``Authentication`` section for more
 details on how to set your credentials) and return a Model object
-that you can use to make local predictions. If you want to use a
-specfic connection object for the remote retrieval, you can set it as second
+that will be stored in the ``./storage`` directory and
+you can use to make local predictions. If you want to use a
+specfic connection object for the remote retrieval or a different storage
```
Comment (jaor) [inline docs/index.rst]: specific

File: docs/index.rst [hunk 3]
```diff
@@ -5597,9 +5800,11 @@ This will retrieve the remote association information, using an implicitly
 built
 ``BigML()`` connection object (see the ``Authentication`` section for more
 details on how to set your credentials) and return an ``Association`` object
-that you can use to extract the rules found in the original dataset.
+that will be stored in the ``./storage`` directory and
+you can use to extract the rules found in the original dataset.
 If you want to use a
-specfic connection object for the remote retrieval, you can set it as second
+specfic connection object for the remote retrieval or a different storage
```
Comment (jaor) [inline docs/index.rst]: specific

File: docs/index.rst [hunk 4]
```diff
@@ -5738,9 +5944,11 @@ This will retrieve the remote topic model information,
 using an implicitly built
 ``BigML()`` connection object (see the ``Authentication`` section for more
 details on how to set your credentials) and return a ``TopicModel``
-object that you can use to obtain local topic distributions.
+object that will be stored in the ``./storage`` directory and
+you can use to obtain local topic distributions.
 If you want to use a
-specfic connection object for the remote retrieval, you can set it as second
+specfic connection object for the remote retrieval or a different storage
```
Comment (jaor) [inline docs/index.rst]: specific

File: docs/index.rst [hunk 5]
```diff
@@ -5817,9 +6025,11 @@ the remote time series information,
 using an implicitly built
 ``BigML()`` connection object (see the ``Authentication`` section for more
 details on how to set your credentials) and return a ``TimeSeries``
-object that you can use to obtain local forecasts.
+object that will be stored in the ``./storage`` directory and
+you can use to obtain local forecasts.
 If you want to use a
-specfic connection object for the remote retrieval, you can set it as second
+specfic connection object for the remote retrieval or a different storage
```
Comment (jaor) [inline docs/index.rst]: specific

File: docs/index.rst [hunk 6]
```diff
@@ -5846,13 +6056,128 @@ local time series object:
...
+    local_pca = PCA(
+        'pca/502fdbcf15526876210042435')
+
+
+This will create an object that stores the remote information that defines
+the PCA, needed to generate
+projections to the new dimensionally reduced components. The remote resource
+is automatically downloaded the first time the the PCA is instantiated by
+using an implicitly built
+``BigML()`` connection object (see the ``Authentication`` section for more
```
Comment (jaor) [inline docs/index.rst]: can you make the ``Authentication`` reference a link to the section?

File: docs/index.rst [hunk 7]
```diff
@@ -5846,13 +6056,128 @@ local time series object:
...
+details on how to set your credentials). The JSON that contains this
+information is stored in a ``./storage`` directory, which is the default
+choice. If you want to use a
+specfic connection object to define the credentials for the authentication
```
Comment (jaor) [inline docs/index.rst]: specific

File: docs/index.rst [hunk 8]
```diff
@@ -5846,13 +6056,128 @@ local time series object:
...
 method in the
 same call (unlike in the standard calls where the number of fields returned is
 limited).
 
 
+Local Projections
+-----------------
+
+Using the local PCA object, you can compute the projection of
+an input data set into the new components:
```
Comment (jaor) [inline docs/index.rst]: dataset (?)

**Comment by mmerce** [comment]
Waiting for some fixes to go to production for the tests to run.

**Pull request closed without merging**